runtime/src/main/kotlin/kotlin/native/Text.kt

```diff
@@ -9,52 +9,103 @@ package kotlin.native
 /**
  * Converts an UTF-8 array into a [String]. Replaces invalid input sequences with a default character.
  */
+@Deprecated("Use decodeToString or decodeZeroTerminatedToString instead")
 public fun ByteArray.stringFromUtf8() : String {
+    @Suppress("DEPRECATION")
     return this.stringFromUtf8(0, this.size)
 }
 
 /**
  * Converts an UTF-8 array into a [String]. Replaces invalid input sequences with a default character.
  */
+@Deprecated("Use decodeToString or decodeZeroTerminatedToString instead")
 public fun ByteArray.stringFromUtf8(start: Int = 0, size: Int = this.size) : String {
-    checkBoundsIndexes(start, start + size, this.size)
-    return stringFromUtf8Impl(start, realSize(this, size))
+    val endIndex = start + size
+    checkBoundsIndexes(start, endIndex, this.size)
+
+    val realSize = realEndIndex(this, start, endIndex) - start
+    return unsafeStringFromUtf8Impl(start, realSize)
 }
 
-@SymbolName("Kotlin_ByteArray_stringFromUtf8")
-internal external fun ByteArray.stringFromUtf8Impl(start: Int, size: Int) : String
+/**
+ * Decodes a string from the bytes in UTF-8 encoding in this array.
+ * Bytes following the first occurrence of `0` byte, if it occurs, are not decoded.
+ *
+ * Malformed byte sequences are replaced by the replacement char `\uFFFD`.
+ */
+@SinceKotlin("1.3")
+public fun ByteArray.decodeZeroTerminatedToString() : String {
```

Comment:
This new API should be discussed

Reply:
The points to discuss:
- Naming
- Placement: `kotlin.text` or `kotlin.native`

Reply:
Or kotlinx.cinterop. It is actually seems mostly used there.

Reply:
Moved to kotlinx.cinterop

Reply:
Named `toKString()`